vanilla-installer/main.py

```diff
@@ -1,35 +1,37 @@
 """
 Most important functions of VanillaInstaller.
 """
-
-PATH_FILE = "data/mc-path.txt"
-FOLDER_LOC = ""
-TEMP_FOLDER = "temp/"
-
-import logging
-
 # IMPORTS
 import os
 import sys
-import zipfile
-import requests
+import logging
+import logging.handlers  # pylance moment
 import webbrowser
 import subprocess
+import pathlib
+import requests
 import minecraft_launcher_lib as mll
 
+
 # LOCAL
 import gui
 import theme
 
+PATH_FILE = str(pathlib.Path("data/mc-path.txt").resolve())
+FOLDER_LOC = ""
+TEMP_FOLDER = str(pathlib.Path("/temp").resolve())
+
+
 def set_dir(path: str) -> str:
     """Sets the Minecraft game directory.
 
     Args:
         path (str): The path to the Minecraft game directory.
     """
     if isinstance(path, str):  # only strings can be written
-        open(PATH_FILE, "w").write(path)
+        open(PATH_FILE, "w", encoding="utf-8").write(path)
         return path
+    return
```

Comment:
Return type is `str`, but you return `None`?

Reply:
Should probably redo this here. Ideally this makes it go back to the previous path if None is returned, but we should probably just rerun mll.utils.get_minecraft_path here so that the path doesn't appear as none.

Reply:
Or you can do `() -> str | None` to keep things clear

Reply:
Alright, done.